### ISSUE TITLE
[Feature] Comparison of products by cost per weight/volume of nutrients

### ISSUE THREAD

**Issue by aleksejrs** [openfoodfacts/openfoodfacts-server#452]
1. Enter prices (and weights/volumes, if applicable) of two or more products/categories
2. Get a table comparing the costs of each nutrient per gram or milliliter (maybe along with the normal quantity comparison if possible).

**Comment by TaciteOFF** [comment]
Prices are a very touchy topic and I think it shouldn't be directly implemented in Open Food Facts but on a second project.
To make our database grow bigger we need to have verified datasets from manufacturers. There's a high chance that manufacturers will never want to help us if we clearly show prices.

See #prices on our Slack for more information.

**Comment by aleksejrs** [comment]
I mean only entering prices in the comparison UI, not storing them.

**Comment by aleksejrs** [comment]
Teolemon also mentioned "how many calories per dollar?"

**Comment by aleksejrs** [comment]
The prices must be easily editable without reloading the page (like in Android's calculator).

**Comment by aleksejrs** [comment]
![nutrient-price-comparison-001](https://cloud.githubusercontent.com/assets/1025861/18241448/42b799f4-7362-11e6-9957-60e0f6b64d3c.png)

An imperfect graphical comparison.

Product A (left) is 3 times cheaper per mass/volume than product B.  The widths mean the relative amounts of the products you an purchase for a unit of money.

Heights of the rectangles are the relative amounts of nutrients in them.

Product A contains less Green nutrient than product B does, but you can get more of it for the same price, if you don't mind the Yellow nutrient and the grey Rest, of which there is more in A than in B.

Red is hard to compare in this case, maybe numbers would help.

**Comment by aleksejrs** [comment]
Of course, it is a client-side feature.  Maybe I shouldn't propose it for the server until it's free, or for the Android apps until they can work offline.

**Comment by aleksejrs** [comment]
https://ru.openfoodfacts.org/product/4607177745602 about 14 roubles for 350 g.
https://ru.openfoodfacts.org/product/4601347002052 about 28 roubles for 400 g.

First column shows how much of a nutritional value you get for 28 ₽, and the second column shows how much of the correspondent value of the other product that is.

Бородинский, 2 halves, 28 ₽ for 700 g:

* p 47,6	1.50
* f 9,1	        0.91
* c 278.6	1.36
* e 5894	1.34


Ароматный, 28 ₽ for 400 g:

* p 31.6	0.66
* f 10	1.09
* c 204.4	0.73
* e 4384	0.74
* sliced (time)
* tasty (subjective)

Things to consider:

* Бородинский lists its carbohydrates as "углеводы усвояемые" (digestible carbohydrates), so those could include fiber while those of ароматный do not, making the difference even bigger?
* Ароматный lists vitamins.
* Ingredients.
* There may be other differences that cannot be found on the label, but may be mentioned in the GOSTs.

**Comment by teolemon** [comment]
@aleksejrs @TaciteOFF @raphael0202 @raphodn https://prices.openfoodfacts.org/app/prices starts solving this.
